src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -762,17 +755,18 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                     // start stream object
                     $objtype = 'stream';
                     $offset += 6;
-                    if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset), $matches)) {
+                    if (1 == preg_match('/^([\r]?[\n])/isU', substr($pdfData, $offset, 4), $matches)) {
                         $offset += \strlen($matches[0]);
                         $pregResult = preg_match(
                             '/(endstream)[\x09\x0a\x0c\x0d\x20]/isU',
-                            substr($pdfData, $offset),
+                            $pdfData,
                             $matches,
```

Comment:
Same case here. Instead of getting potentially huge substring just to feed it to regexp we can just use preg_match's offset parameter.

But using offset parameter changes offset arithmetics. So, we gonna repair it the nex few lines.